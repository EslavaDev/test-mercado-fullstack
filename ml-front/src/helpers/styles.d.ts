/* eslint-disable @typescript-eslint/no-empty-interface */
import 'styled-components';
import { styledTheme } from './theme-style-components';

type AppTheme = typeof styledTheme;

declare module 'styled-components' {
  export interface DefaultTheme extends AppTheme {}
}

declare global {
  type FontWeight = 'bold' | 'normal' | 'regular' | 700;
}
