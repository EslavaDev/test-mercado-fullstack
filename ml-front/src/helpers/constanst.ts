// SIZE CONSTANTS
export const MARGIN_ML = 32;
export const PADDING_ML = 16;

// FONT SIZE CONSTATNS

export const VERY_TINY = 12;
export const TINY = 14;
export const SMALL = 16;
export const REGULAR = 24;
export const NORMAL = 28;
export const BIG = 46;
