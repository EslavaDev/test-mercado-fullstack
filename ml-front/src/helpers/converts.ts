export const kelvinToCelcius = (value: number) => value - 273.15;
