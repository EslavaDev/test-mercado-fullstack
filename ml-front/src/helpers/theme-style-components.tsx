export const styledTheme = {
  colors: {
    WHITE: '#FFFFFF',
    YELLOW: '#FFE600',
    BLUE: '#3483FA',
    BLACK: '#333333',
    SNOW: '#EEEEEE',
    LIGTH_GRAY: '#999999',
    GRAY: '#666666',
  },
};
