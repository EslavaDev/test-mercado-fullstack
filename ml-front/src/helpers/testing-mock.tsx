import React from 'react';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components';
import { AppStore } from '../store';
import { styledTheme } from './theme-style-components';

interface Props {
  children: React.ReactNode;
}

export const TestingMock = ({ children }: Props) => {
  return (
    <ThemeProvider theme={styledTheme}>
      <Provider store={AppStore}>{children}</Provider>
    </ThemeProvider>
  );
};
