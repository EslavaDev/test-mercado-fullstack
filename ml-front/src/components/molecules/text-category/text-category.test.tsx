import * as React from 'react';
import { render, cleanup } from '@testing-library/react';
import { TextCategory } from '.';
import { TestingMock } from '../../../helpers/testing-mock';

afterEach(cleanup);

describe('TextCategory', () => {
  it('works', () => {
    const mockData = {
      isLast: false,
      text: 'mockData',
    };
    const { container } = render(
      <TestingMock>
        <TextCategory {...mockData} />
      </TestingMock>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
  it('isLast is true', () => {
    const mockData = {
      isLast: false,
      text: 'mockData',
    };
    const { getByText } = render(
      <TestingMock>
        <TextCategory {...mockData} />
      </TestingMock>,
    );
    expect(getByText(/>/)).toBeInTheDocument();
    expect(getByText(/mockData/)).toBeInTheDocument();
  });
  it('isLast is false', () => {
    const mockData = {
      isLast: true,
      text: 'mockData',
    };
    const { getByText } = render(
      <TestingMock>
        <TextCategory {...mockData} />
      </TestingMock>,
    );
    expect(getByText(/mockData/)).toBeInTheDocument();
  });
});
