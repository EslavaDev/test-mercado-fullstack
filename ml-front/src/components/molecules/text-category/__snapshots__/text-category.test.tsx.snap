// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextCategory works 1`] = `
<div
  class="sc-bdfBwQ ca-DIka"
>
  <p
    class="sc-dlfnbm fAqtVb"
  >
    mockData
  </p>
   
  <span
    class="sc-gsTCUz gUGYDt"
  >
     
    &gt;
     
  </span>
</div>
`;
